### PR TITLE
feat: replace gomnd by mnd

### PR DESCRIPTION
**Pull request by ldez**
Rename gomnd to mnd:
- Deprecate gomnd.
- Add mnd.

This PR also fixes the problem of consistency between the analyzer name and linter name like #4567.

<details><summary>Before</summary>

```console
$ golangci-lint run 
main.go:28:23: mnd: Magic number: 200, in <condition> detected (gomnd)
        if res.StatusCode != 200 {
                             ^
main.go:21:12: mnd: Magic number: 2, in <assign> detected (gomnd)
                Timeout: 2 * time.Second,
                         ^
```

</details>

<details><summary>With the PR</summary>

```console
$ golangci-lint run 
main.go:28:23: Magic number: 200, in <condition> detected (mnd)
        if res.StatusCode != 200 {
                             ^
main.go:21:12: Magic number: 2, in <assign> detected (mnd)
                Timeout: 2 * time.Second,
                         ^
```

</details>

### CODE REVIEW

**Review by bombsimon — APPROVED** [review]
LGTM, as long as the name actually _is_ just `mnd`. I see it's set to `mnd` in the code but most of the references refer to it as `go-mnd` and so does the gopher image in the repo.

@tommy-muehle Can you confirm that the name of the linter should be just `mnd` or would you rather set the name to `gomnd` on the analyzer? I guess `mnd` makes more sense since it's the name of the binary as well but might as well ask to clarify.

**Comment by ldez** [comment]
> I see it's set to mnd in the code but most of the references refer to it as go-mnd and so does the gopher image in the repo.

So you are saying that only my first commit (coming from the previous PR) was enough, it's because you need more PRs to review? :smile_cat:

**Comment by ldez** [comment]
> I'm sorry if this yielded extra work for you. 

I was just kidding :smile_cat: 

I was torn between the name change and the deprecation, so I used your comment as justification.

**Comment by ldez** [comment]
More seriously, the readme is a mix of `go-mnd` and `mnd` but the command, the binary, the analyzer, and the brew-tap are called `mnd`.
Only the Docker image and the repository are named `go-mnd`.

We will wait for the feedback of the author but `mnd` seems to be the right name.

**Comment by bombsimon** [comment]
> We will wait for the feedback of the author but `mnd` seems to be the right name.

Yeah 100% agree! I pinged to see if we got some response within a day or two max but I wouldn't block on it.

File: pkg/lint/lintersdb/builder_linter.go
Comment (Antonboom) [inline pkg/lint/lintersdb/builder_linter.go:297]: Why for `goerr13` you used alternative name without deprecation, but here you deprecated the old name? oO

File: pkg/golinters/mnd.go
Comment (Antonboom) [inline pkg/golinters/mnd.go:11]: I thought MND means "monad" (like in functional programming) :D 
really not obvious that mnd is for "magic number detector"

why not use name like "magic-number"?

**Review by Antonboom — CHANGES_REQUESTED** [review]

File: pkg/lint/lintersdb/builder_linter.go
Comment (ldez) [reply pkg/lint/lintersdb/builder_linter.go:297]: https://github.com/golangci/golangci-lint/pull/4562

File: pkg/golinters/mnd.go
Comment (ldez) [reply pkg/golinters/mnd.go:11]: We will not change the name of linters, golangci-lint is a wrapper.

File: pkg/lint/lintersdb/builder_linter.go
Comment (Antonboom) [reply pkg/lint/lintersdb/builder_linter.go:297]: but why not `.WithAlternativeNames("gomd").` here?
I try to understand the reason of approaches difference

Comment (ldez) [reply pkg/lint/lintersdb/builder_linter.go:297]: https://github.com/golangci/golangci-lint/pull/4572/commits/623d457238670ee79767f1dcd63f5e9b82e0b519#diff-23eb637b7fd1abddf7f975546966746b84c387b8294f08c9a9a19cd7dee59082R12-R19

**Review by Antonboom — APPROVED** [review]

**Pull request merged**
